src/tdydm.c

```diff
@@ -114,6 +114,13 @@ PetscErrorCode TDyCreateDM(DM *dm) {
     ierr = DMPlexCreateBoxMesh(PETSC_COMM_WORLD, dim, PETSC_FALSE, faces,
                                lower, upper, NULL, PETSC_TRUE, dm);
     CHKERRQ(ierr);
+
+    int num_cell_in_set;
+    num_cell_in_set = Nx * Ny * Nz;
```

Comment:
Nx, Ny, Nz are the global values.  What happens in parallel when each process sets the label value for all cells?

Reply:
The solution is to wrap these line within:

   PetscInt rank;
    ierr = MPI_Comm_rank(PetscObjectComm((PetscObject)*dm),&rank);
    if (!rank) {
      ...
    }

so that only process 0 sets the labels.  This is what was causing master to fail.  When I wrap the setting of the label, this works for v3.13 and the tip of master.